src/core/ext/client_channel/client_channel.c

```diff
@@ -676,6 +677,9 @@ typedef struct client_channel_call_data {
   grpc_linked_mdelem lb_token_mdelem;
 } call_data;
 
+static void apply_final_configuration_locked(grpc_exec_ctx *exec_ctx,
```

Comment:
How about just moving the function definition up here, so that we don't need the forward declaration?